### ISSUE TITLE
Include AssertJ with Quarkus releases

### ISSUE THREAD

**Issue by ghost** [quarkusio/quarkus#38689]
### Description

AssertJ is a widely-used library for fluent assertions in Java, commonly used in conjunction with testing frameworks such as JUnit — which Quarkus uses by default.

While built-in JUnit assertions do the job for the most part, AssertJ provides a rich set of assertions and truly helpful error messages.

The ask in this issue is for you to consider including AssertJ with Quarkus releases so there is no need to explicitly specify the version number. This would ensure that a given, tested version of AssertJ is released alongside the other libraries included in the Maven BOM.

Another (even better!) approach could be to include it as part of `io.quarkus:quarkus-junit5`, but I'm not sure about the scope for that artifact.

### Benefits:

- Simplification of dependency management by reducing the need to specify version numbers
- Ensures compatibility and stability by providing a tested version of AssertJ alongside other libraries
- Streamlines the development process by promoting consistency in dependency management across different artifacts

**Comment by geoand** [comment]
I think we used to include it in the BOM, right @gsmet?

**Comment by lasteris** [comment]
@geoand Maybe you should close this issue ? I've just now observed that in 3.12.3 you have access to assertj if you add
testImplementation 'org.assertj:assertj-core:3.26.3'
(it is not working if you are trying to add version below 3.26.3, but as i assume thats correct behaviour, reference https://github.com/quarkusio/quarkus/pull/41799)

also, user who created this issue is DELETED :(

**Comment by geoand** [comment]
Good point @lasteris

**Comment by beatngu13** [comment]
I can see, e.g., Hamcrest in the [BOM](https://github.com/quarkusio/quarkus/blob/ffafe44e2f4aa5e20b026df9d3334164a5249ff5/bom/application/pom.xml), but (still) no AssertJ?

>I've just now observed that in 3.12.3 you have access to assertj if you add
testImplementation 'org.assertj:assertj-core:3.26.3'

But then you manage the version yourself?

**Comment by geoand** [comment]
> I can see, e.g., Hamcrest in the [BOM](https://github.com/quarkusio/quarkus/blob/ffafe44e2f4aa5e20b026df9d3334164a5249ff5/bom/application/pom.xml), but (still) no AssertJ?

This is probably due to RestAssured needing it

**Comment by beatngu13** [comment]
>This is probably due to RestAssured needing it

Yes, indeed.

Would it still make sense for you to include AssertJ in the BOM? If so, there would be two possible dependencies:

1. [`org.assertj:assertj-core`](https://mvnrepository.com/artifact/org.assertj/assertj-core/3.26.3): comes only with [AssertJ Core](https://assertj.github.io/doc/#assertj-core).
2. [`org.assertj:assertj-bom`](https://mvnrepository.com/artifact/org.assertj/assertj-bom/3.26.3): manages both [AssertJ Core](https://assertj.github.io/doc/#assertj-core) and [AssertJ Guava](https://assertj.github.io/doc/#assertj-guava) (which has a [provided dependency](https://mvnrepository.com/artifact/org.assertj/assertj-guava/3.26.3) on Guava itself).

**Comment by geoand** [comment]
I'm +0. @gsmet ?

**Comment by JohnnyXavier** [comment]
I was just looking for **_assertJ+quarkus_** and found this issue and also the reason it was removed back in the day:

[quarkus-2-8-0-final-released](https://quarkus.io/blog/quarkus-2-8-0-final-released/)

1st comment by @gsmet :

> Assertj out of the BOM
> First things first: if you are using Assertj in your project, we moved it out of our BOM as binary compatibility was broken quite often and it made things hard for some projects.
>
> So, from now on, define the version in your projects. The latest version is 3.22.0.

**Comment by scordio** [comment]
AssertJ maintainer here 👋 

I'm chiming in as one AssertJ user recently asked why Spring Boot delivers AssertJ out of the box and Quarkus doesn't, and I ended up in this issue.

> we moved it out of our BOM as binary compatibility was broken quite often

Indeed, that was an unfortunate issue we caused more than once years ago, but [guardrails](https://github.com/assertj/assertj/pull/2271) were introduced once we realized it, and we have had many binary-compatible releases since then.

If you'd like to reconsider this topic and have any questions, feel free to reach out to me!
